tritond/mock.go

```diff
@@ -37,6 +38,26 @@ func (c *MockClient) Close(ctx context.Context) error {
 	return nil
 }
 
+// Reset resets MockClient to the initial state
+func (c *MockClient) Reset() {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	c.StreamData = make(map[string]([](map[string]interface{})))
+	c.PartitionCount = make(map[string]int)
+}
+
+func (c *MockClient) unlock() {
+	c.lock.Unlock()
+}
+
+// PrepareForAssert prepares MockClient to be asserted when it was used in async code.
+// Test code *must* defer on result of this method
+func (c *MockClient) PrepareForAssert() func() {
+	runtime.Gosched()
+	c.lock.Lock()
+	return c.unlock
```

Comment:
Personal preference would be to use an anonymous function here so it can be very specific to this use case.

Reply:
Also, not sure if you really want `runtime.Gosched()`. It looks like that forces a context switch, which might not be what you want? 
https://stackoverflow.com/questions/13107958/what-exactly-does-runtime-gosched-do

Reply:
The idea is to make it reusable for all cases when clients want to test async usages of this code. Anonymous func in my code is not reusable.